fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -0,0 +1,274 @@
+                srIcebergTable.getIcebergTable(), snapshot.get(),
+                icebergPredicates, true).planFiles()) {
+            DataFile file = task.file();
+            LOG.info("Scan with file " + file.path() + ", file record count " + file.recordCount());
+
+            TScanRangeLocations scanRangeLocations = new TScanRangeLocations();
+
+            THdfsScanRange hdfsScanRange = new THdfsScanRange();
+            hdfsScanRange.setFull_path(file.path().toString());
+            hdfsScanRange.setOffset(task.start());
+            hdfsScanRange.setLength(task.length());
+            // For iceberg table we do not need partition id
+            hdfsScanRange.setPartition_id(-1);
+            hdfsScanRange.setFile_length(file.fileSizeInBytes());
+            hdfsScanRange.setFile_format(IcebergUtil.getHdfsFileFormat(file.format()).toThrift());
+            TScanRange scanRange = new TScanRange();
+            scanRange.setHdfs_scan_range(hdfsScanRange);
+            scanRangeLocations.setScan_range(scanRange);
+
+            // TODO: get hdfs block location information for scheduling, use iceberg meta cache
+            TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
+            scanRangeLocations.addToLocations(scanRangeLocation);
+
+            result.add(scanRangeLocations);
+        }
+    }
+
+    @Override
+    protected String debugString() {
+        MoreObjects.ToStringHelper helper = MoreObjects.toStringHelper(this);
+        helper.addValue(super.debugString());
+        helper.addValue("icebergTable=" + srIcebergTable.getName());
+        return helper.toString();
+    }
+
+    @Override
+    protected String getNodeExplainString(String prefix, TExplainLevel detailLevel) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(srIcebergTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
+        if (!conjuncts.isEmpty()) {
+            output.append(prefix).append("PREDICATES: ").append(
+                    getExplainString(conjuncts)).append("\n");
+        }
+
+        output.append(prefix).append(String.format("cardinality=%s", cardinality));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("avgRowSize=%s", avgRowSize));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
+        output.append("\n");
+
+        return output.toString();
+    }
+
+    @Override
+    protected String getNodeVerboseExplain(String prefix) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(srIcebergTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
```

Comment:
Do we support sort-column for external iceberg table in starrocks ?

Reply:
This is only used for'order by' which is different from the sorting column in the iceberg